### PR TITLE
Deprecation Updates for SceneTransforms Methods

### LINKED ISSUE #12059: Remove `SceneTransforms.wgs84To*` functions in 1.121
The following functions were deprecated in https://github.com/CesiumGS/cesium/pull/12000, and should be entirely removed for the 1.121 release:

- `SceneTransforms.wgs84ToDrawingBufferCoordinates` has been deprecated. It will be removed in 1.121. Use `SceneTransforms.worldToDrawingBufferCoordinates` instead.
- `SceneTransforms.wgs84ToWindowCoordinates` has been deprecated. It will be removed in 1.21. Use `SceneTransforms.worldToWindowCoordinates` instead.

### PR DESCRIPTION
**Pull request by jvrjsanchez**
# Description

This pull request addresses the deprecation of certain methods within the `SceneTransforms` module. Specifically, the methods `wgs84ToDrawingBufferCoordinates` and `wgs84ToWindowCoordinates` are being deprecated and will be removed in future versions. This update ensures that our codebase aligns with the latest changes and improves future maintainability.

- The updated methods provide improved functionality and are part of the new design. The transition to these methods should enhance consistency and performance.
- Ensured and tested thoroughly to confirm that the new methods behave as expected and integrated seamlessly with existing code.

- Updated method calls in the codebase from `wgs84ToDrawingBufferCoordinates` to `worldToDrawingBufferCoordinates`.
- Updated method calls in the codebase from `wgs84ToWindowCoordinates` to `worldToWindowCoordinates`.
PNG image
Test cases passed.

## Issue number and link 

fixes #12059 

## Testing plan

- Ran tests using the original `gulpfile.js` to execute them.
- Verified that all tests for `SceneTransforms` pass successfully in different viewing modes (3D, ColumbusView, 2D) and frustum configurations.
- Ensured that test output is correctly reported and errors are handled gracefully.

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have updated the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,8 @@
   ],
   "dependencies": {
     "@cesium/engine": "^10.1.0",
-    "@cesium/widgets": "^7.1.0"
+    "@cesium/widgets": "^7.1.0",
+    "cesium": "^1.120.0"
```
Comment (ggetz) [inline package.json:56]: Please remove this unrelated change.

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+#### @cesium/engine
+
+##### Additions :Deprecation Updates for SceneTransforms Methods:
+
+- Updated method calls in the codebase from `wgs84ToDrawingBufferCoordinates` to `worldToDrawingBufferCoordinates`.
+- Updated method calls in the codebase from `wgs84ToWindowCoordinates` to `worldToWindowCoordinates`.
+- Ran tests using the original `gulpfile.js` to execute them.
+- Verified that all tests for `SceneTransforms` pass successfully in different viewing modes (3D, ColumbusView, 2D) and frustum configurations.
+- Ensured that test output is correctly reported and errors are handled gracefully.
+
```
Comment (ggetz) [inline CHANGES.md:14]: Changes should all go under the next scheduled release, `### 1.121 - 2024-09-01`.

Comment (ggetz) [reply CHANGES.md:14]: I would replace this all with:

```
- `SceneTransforms.wgs84ToWindowCoordinates` has been removed. Use `SceneTransforms.worldToWindowCoordinates` instead.
- `SceneTransforms.wgs84ToDrawingBufferCoordinates` has been removed. Use `SceneTransforms.worldToDrawingBufferCoordinates` instead.
```

and put it under the 

`##### Breaking Changes :mega:` 

section.

**Review by ggetz — CHANGES_REQUESTED** [review]
Great, thanks @jvrjsanchez! I have a few comments on the code that should be addressed before we can merge this.

**Comment by jjspace** [comment]
@jvrjsanchez Our monthly release happens on Tuesday and this needs to make it into that. In the interest of time because we have a long weekend I made some final touches on this to get it over the finish line. Thanks for your work on the PR.

**Review by ggetz — APPROVED** [review]

**Comment by ggetz** [comment]
Thanks all!

**Pull request merged**